test/sender.test.js

```diff
@@ -25,25 +25,112 @@ const proxyOptions = {
-    y: '__ptaol41JWSpTTL525yVEfzmY8A6Vi_QrW1FjKcHMg'
 }
-const JWK = {
-    ...PUBLIC_KEY,
-    kid: 'testapp',
-    kty: 'EC',
-    d: PRIVATE_KEY,
-    crv: 'P-256',
+
+const USER_NAME = 'testapp';
+const PRIVATE_KEY = '9b9x5WhJywDEuo1KGQWSPNxtX-6X6R2BRCKhYMMY6n8';
+const AUTH = {
+    kid: USER_NAME,
+    d: PRIVATE_KEY
 }
 
 async function sleep(ms) {
     return new Promise(resolve => setTimeout(resolve, ms));
 }
 
+describe('Sender auth config checks suite', function () {
+    it('requires a username for authentication', async function () {
+        try {
+            new Sender({
+                bufferSize: 512,
+                auth: {
+                    d: 'privateKey'
+                }
+            });
+            fail('it should not be able to create the sender');
+        } catch(err) {
+            expect(err.message).toBe('Missing username, please, specify the \'kid\' property of the \'auth\' config option. ' +
+                'For example: new Sender({auth: {kid: \'username\', d: \'private key\'}})');
+        }
+    });
+
+    it('requires a non-empty username', async function () {
+        try {
+            new Sender({
+                bufferSize: 512,
+                auth: {
+                    kid: '',
+                    d: 'privateKey'
+                }
+            });
+            fail('it should not be able to create the sender');
+        } catch(err) {
+            expect(err.message).toBe('Missing username, please, specify the \'kid\' property of the \'auth\' config option. ' +
+                'For example: new Sender({auth: {kid: \'username\', d: \'private key\'}})');
+        }
+    });
+
+    it('requires that the username is a string', async function () {
+        try {
+            new Sender({
+                bufferSize: 512,
+                auth: {
+                    kid: 23,
+                    d: 'privateKey'
+                }
+            });
+            fail('it should not be able to create the sender');
+        } catch(err) {
+            expect(err.message).toBe('Please, specify the \'kid\' property of the \'auth\' config option as a string. ' +
+                'For example: new Sender({auth: {kid: \'username\', d: \'private key\'}})');
+        }
+    });
+
+    it('requires a private key for authentication', async function () {
+        try {
+            new Sender({
+                auth: {
+                    kid: 'username'
+                }
+            });
+            fail('it should not be able to create the sender');
+        } catch(err) {
+            expect(err.message).toBe('Missing private key, please, specify the \'d\' property of the \'auth\' config option. ' +
+                'For example: new Sender({auth: {kid: \'username\', d: \'private key\'}})');
+        }
+    });
+
+    it('requires a non-empty private key', async function () {
+        try {
+            new Sender({
+                auth: {
+                    kid: 'username',
```

Comment:
In Java we call these keyId and token. It might be a good idea to use these field names here too.